src/main/java/core/basesyntax/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(List<FruitTransaction> transactions, Map<String, Integer> inventory) {
+        for (FruitTransaction transaction : transactions) {
+            if (transaction.getOperation() == FruitTransaction.Operation.BALANCE) {
+                inventory.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Ensure that the `transaction.getFruit()` and `transaction.getQuantity()` methods return valid values. Consider adding validation or logging to handle unexpected or invalid data gracefully.